tools/refguide_check.py

```diff
@@ -809,6 +852,40 @@ def check_doctests_testfile(fname, verbose, ns=None,
 
     return results
 
+def iter_included_files(base_path, args, suffixes=('.py', '.rst')):
+    if os.path.exists(base_path) and os.path.isfile(base_path):
+        yield base_path
+    for dir_name, subdirs, files in os.walk(base_path, topdown=True):
+        if dir_name in RST_SKIPLIST:
+            if args.verbose > 0:
+                sys.stderr.write('skipping files in %s' % dir_name)
+            files = []
+        for p in RST_SKIPLIST:
+            if p in subdirs:
+                if args.verbose > 0:
+                    sys.stderr.write('skipping %s and subdirs' % p)
+                subdirs.remove(p)
+        for f in files:
+            if (os.path.splitext(f)[1] in suffixes and
+                    f not in RST_SKIPLIST):
+                yield os.path.join(dir_name, f)
+
+def check_rst(base_path, results, args, dots):
+    for filename in iter_included_files(base_path, args):
```

Comment:
I'm a little puzzeld why `check_rst` is also matching `.py` files. Am I missing something?

Reply:
Any suggestions for better names? `check_file`?

Reply:
Doesn't this result in the python files being checked twice, once through `check_rst(suffixes='.py')`, and once through one of `check_items` / `check_rest` / `check_doctests`?

My point in that comment was I no longer understand what these functions are checking. A docstring might help.

Reply:
add docstrings